### PR TITLE
Fix pretraining with iterable/streaming Dataset

### PR DESCRIPTION
**Pull request by jphme**
Due to various changes pretraining_dataset didn't work anymore, this should fix it (using it without problems with a streaming dataset, works for local + remote).

### CODE REVIEW

**Comment by NanoCode012** [comment]
What would happen if max_steps is set to a really large number? Would it default to only running through the entire dataset? I'm a bit curious how to determine this value.

**Comment by jphme** [comment]
> What would happen if max_steps is set to a really large number? Would it default to only running through the entire dataset? I'm a bit curious how to determine this value.

No, it would run max_steps (and determine the number of epochs accordingly), see https://github.com/huggingface/transformers/blob/6acc27eea853885270dba5313181443d43e31f2c/src/transformers/trainer.py#L1605 .

For IterableDatasets, max_steps is necessary (see https://github.com/huggingface/transformers/blob/6acc27eea853885270dba5313181443d43e31f2c/src/transformers/trainer.py#L574 ).

**Comment by kostum123** [comment]
Does this pr fix finetune with raw corpus? 
dataset type:
completion: raw corpus
{"text": "..."}

File: src/axolotl/utils/data.py [hunk 1]
```diff
@@ -637,6 +640,12 @@ def load_pretraining_dataset(path, tokenizer, max_tokens=2048, seed=42):
     encode = functools.partial(encode_pretraining, tokenizer, max_tokens)
     dataset = load_dataset(path, streaming=True, split="train")
     dataset = dataset.shuffle(seed=seed, buffer_size=10_000)
-    # TODO dynamically figure out which columns/features to remove
```
Comment (winglian) [inline src/axolotl/utils/data.py:640]: can you leave this TODO please? some datasets have additional features that need to be dropped

Comment (winglian) [reply src/axolotl/utils/data.py:640]: oh, nvm, I see now, map w/ input_columns just passes that so the rest aren't even present 👍

**Review by winglian — APPROVED** [review]
lgtm. thanks!

**Pull request merged**